File: prototype/TextRenderer.cs
namespace prototype.GUI
{
    using System;
    using System.Collections.Generic;
    using static SDL2.SDL;

    internal class TextRenderer
    {
        //TODO parameterize width/height to enable loading different png dimensions/ratios

        private readonly TextureLoader textureLoader_;
        private TextureWrapper letters_;

        

        public TextRenderer(TextureLoader textureLoader)
        {
            textureLoader_ = textureLoader;
        }

        public void Initialize(IntPtr renderer)
        {
            letters_ = textureLoader_.LoadTexture("Resources/letters.png", renderer);
        }

        public void Render(string text)
        {
            //TODO escape sequences and other stuff
            for (var i = 0; i < text.Length; i++)
            {
                var character = text[i];
                letters_.Render(i * LetterClips.LetterWidth * 2, 0, LetterClips.GetClip(character) , 2);
            }
        }
    }
}

File: src/prototype/textrenderer.py
from prototype import texture
from prototype import letterclips
from sdl2 import SDL_Color


class TextRenderer(object):
    def __init__(self, options):
        self._options = options

    def initialize(self, renderer):
        self._letters = texture.load_texture(
            b"../resources/letters.png", # TODO make configurable
            renderer,
            SDL_Color(0xff, 0, 0xdc))

    def render(self, text, target_area, color=None):
        if color:
            self._letters.set_color_mod(color)
        for i, c in enumerate(text):
            self._letters.render(
                target_area.x + i * self._options.scaled_letter_width,
                target_area.y,
                letterclips.get_clip(c),
                self._options.scale)
